tests/ui_tools/test_boxes.py

```diff
@@ -41,8 +42,48 @@ def test_init(self, write_box):
     ])
     def test_generic_autocomplete_no_prefix(self, mocker, write_box, text,
                                             state):
+        write_box.view.set_typeahead_footer = mocker.patch(
+                                'zulipterminal.ui.View.set_typeahead_footer')
         return_val = write_box.generic_autocomplete(text, state)
         assert return_val == text
+        write_box.view.set_typeahead_footer.assert_not_called()
+
+    @pytest.mark.parametrize('text, state, footer_text, wrapping_text', [
+        # mentions
+        ('@Human', 0, ['Human Myself', 'Human 1', 'Human 2'], '@**{}**'),
+        ('@Human', 1, ['Human Myself', 'Human 1', 'Human 2'], '@**{}**'),
+        ('@Group', 0, ['Group 1', 'Group 2', 'Group 3', 'Group 4'], '@*{}*'),
+        ('@Group', 1, ['Group 1', 'Group 2', 'Group 3', 'Group 4'], '@*{}*'),
+        ('@_Human', 0, ['Human Myself', 'Human 1', 'Human 2'], '@_**{}**'),
+        ('@Human', None, ['Human Myself', 'Human 1', 'Human 2'], '@**{}**'),
+        ('@_Human', None, ['Human Myself', 'Human 1', 'Human 2'], '@_**{}**'),
+        ('@NoMatch', None, [], '@**{}**'),
+        ('@_NoMatch', None, [], '@_**{}**'),
+        # streams
+        ('#Stream', 0, ['Stream 1', 'Stream 2', 'Secret stream',
+                        'Some general stream'], '#**{}**'),
+        ('#Stream', 1, ['Stream 1', 'Stream 2', 'Secret stream',
+                        'Some general stream'], '#**{}**'),
+        ('#Stream', None, ['Stream 1', 'Stream 2', 'Secret stream',
+                           'Some general stream'], '#**{}**'),
+        ('#NoMatch', None, [], '#**{}**'),
+        # emojis
+        (':smi', 0, ['smile', 'smiley', 'smirk'], ':{}:'),
+        (':smi', 2, ['smile', 'smiley', 'smirk'], ':{}:'),
+        (':smi', None, ['smile', 'smiley', 'smirk'], ':{}:'),
+        (':NoMatch', None, [], ':{}:'),
+        ])
+    def test_generic_autocomplete_set_footer(self, mocker, write_box,
+                                             wrapping_text, text,
+                                             state, footer_text):
+        write_box.view.set_typeahead_footer = mocker.patch(
+                                'zulipterminal.ui.View.set_typeahead_footer')
+        suggestion = footer_text[state] if state is not None else None
+        footer_text = generate_typeaheads(footer_text, wrapping_text)
```

Comment:
These two lines are to enable calculation of the expected result? If so, maybe move them into a block with the assert?